include/cpp-statsd-client/UDPSender.hpp

```diff
@@ -161,27 +163,33 @@ inline UDPSender::~UDPSender() {
         return;
     }
 
-    if (m_batching) {
+    // If we're running a background thread tell it to stop
+    if (m_batchingThread.joinable()) {
         m_mustExit.store(true, std::memory_order_acq_rel);
         m_batchingThread.join();
     }
 
+    // Cleanup the socket
     close(m_socket);
 }
 
 inline void UDPSender::send(const std::string& message) noexcept {
     m_errorMessage.clear();
+
     // If batching is on, accumulate messages in the queue
-    if (m_batching) {
+    if (m_batchsize > 0) {
```

Comment:
if we are batching we need to enqueue the stat rather than send it right away, regardless of whether that is for the background thread or for the caller to manually flush it later